src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,95 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Make this class immutable. See requirements in the README file
  */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car implements Cloneable {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        List<Wheel> newWheels = new ArrayList<>();
+        for (Wheel oneWheel : wheels) {
+            newWheels.add(oneWheel.clone());
+        }
+        this.wheels = newWheels;
+        this.engine = (engine == null) ? null : new Engine(engine.getHorsePower(),
```

Comment:
let's move it to the separate private method

```suggestion
        this.engine = (engine == null) ? null : engine.clone()
```

Reply:
done